src/Http/Http.Extensions/src/RequestDelegateFactoryOptions.cs

```diff
@@ -36,4 +36,14 @@ public sealed class RequestDelegateFactoryOptions
     /// The list of filters that must run in the pipeline for a given route handler.
     /// </summary>
     public IReadOnlyList<Func<RouteHandlerContext, RouteHandlerFilterDelegate, RouteHandlerFilterDelegate>>? RouteHandlerFilterFactories { get; init; }
+
+    /// <summary>
+    /// The default endpoint metadata to add as part of the creation of the <see cref="RequestDelegateResult.RequestDelegate"/>.
+    /// </summary>
+    /// <remarks>
+    /// This metadata will be included in <see cref="RequestDelegateResult.EndpointMetadata" /> after any metadata inferred during creation of the
+    /// <see cref="RequestDelegateResult.RequestDelegate"/> but before any metadata provided by types in the delegate signature that implement
+    /// <see cref="IEndpointMetadataProvider" /> or <see cref="IEndpointParameterMetadataProvider" />.
+    /// </remarks>
+    public IReadOnlyList<object>? DefaultEndpointMetadata { get; init; }
```

Comment:
What about?

```suggestion
    public IEnumerable<object>? InitialEndpointMetadata { get; init; }
```

I don't see how it's a default given that it needs to be supplied every call to RDF.Create() to be used. I also like `IEnumerable` since that's all we need and it's more likely to be provided by the developer than read.

Reply:
Changing this to `AdditionalEndpointMetadata` as it's not used to initialize the metadata collection anymore, but instead is added at the end of any metadata inferred by RDF, in order to allow the caller to override any inferred metadata.

Reply:
I don't understand this. I think we should stick with `InitialEndpointMetadata`. The caller can already override any inferred metadata by doing whatever it wants with `RequestDelegateResult.EndpointMetadata`. If the caller wants to override inferred metadata, they're free to append their metadata after they copy the `EndpointMetadata` to `EndpointBuilder.Metadata` or just ignore the returned `EndpointMetadata` altogether.

I get wanting attributes to override inference. This is partly why I think it's the `RequestDelegateFactory` itself that should be adding the route handlers' attributes as metadata. Of course, EndpointRouteBuilderExtensions could also just continue to add the attributes to `EndpointBuilder.Metadata` directly as it did before, but I think it's cleaner to have all the metadata in `RequestDelegateResult.EndpointMetadata`.